### PR TITLE
FISH-8436 FISH-8448 FISH-8450 Fix bugs

### PR DESCRIPTION
**Pull request by kalinchan**
Reworked how files are zipped.

Fail gracefully when no collectors are active

Fix issue where exception is thrown when different domain is active than targeted domain

### CODE REVIEW

File: src/main/java/fish/payara/extras/diagnostics/collection/CollectorService.java [hunk 1]
```diff
@@ -165,32 +181,16 @@ public int executeCollection() {
      * @param zipOutputStream
```
Comment (Pandrex247) [inline src/main/java/fish/payara/extras/diagnostics/collection/CollectorService.java:181]: Adjust the javadoc please - you've edited the parameters

File: src/main/java/fish/payara/extras/diagnostics/collection/CollectorService.java [hunk 2]
```diff
@@ -127,30 +140,33 @@ private void init() {
...
 
-        File file = new File((String) parameterMap.get(ParamConstants.DIR_PARAM));
+        Path filePath = Paths.get((String) parameterMap.get(DIR_PARAM));
 
-        try (FileOutputStream fileOutputStream = new FileOutputStream(file.getAbsolutePath() + ".zip")) {
-            ZipOutputStream zipOutputStream = new ZipOutputStream(fileOutputStream);
-            compressDirectory(file, file.getName(), zipOutputStream);
+        ZipOutputStream zipOutputStream = null;
+        try (FileOutputStream fileOutputStream = new FileOutputStream(filePath.toFile() + ".zip")) {
+            zipOutputStream = new ZipOutputStream(fileOutputStream);
```
Comment (Pandrex247) [inline src/main/java/fish/payara/extras/diagnostics/collection/CollectorService.java:161]: I don't see a reason for this variable to be declared outside of the try block
```suggestion
        try (FileOutputStream fileOutputStream = new FileOutputStream(filePath.toFile() + ".zip")) {
            ZipOutputStream zipOutputStream = new ZipOutputStream(fileOutputStream);
```

**Review by Pandrex247 — COMMENTED** [review]
A couple of comments

File: src/main/java/fish/payara/extras/diagnostics/collection/CollectorService.java (hunk 1 above)
Comment (kalinchan) [reply src/main/java/fish/payara/extras/diagnostics/collection/CollectorService.java:181]: resolved in #21

File: src/main/java/fish/payara/extras/diagnostics/collection/CollectorService.java (hunk 2 above)
Comment (kalinchan) [reply src/main/java/fish/payara/extras/diagnostics/collection/CollectorService.java:161]: resolved in #21

**Review by Pandrex247 — APPROVED** [review]

**Pull request merged**